### DISCUSSION TITLE
Is there a way toelegantly,  install *.dmg on macos?

### DISCUSSION THREAD

**Discussion by gcarreno** [community/community#26453] [Actions]
<p>Hi there,</p>
<p>I’m trying to complete an action that install Lazarus/FPC on macos.</p>
<p>From what I’ve investigated, via the command line, you have to:</p>
<ol>
<li>Mount the *.dmg file</li>
<li>Inside the folder it was mounted find a folder or a package</li>
<li>If folder, copy to /Applications</li>
<li>If package the run install package</li>
</ol>
<p>Is there anything for Typesript that could make this available via core or tool-cache?</p>
<p>Cheers,</p>
<p>Gus</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/gcarreno">@gcarreno</a> ,</p>
<p>I found your “<a href="https://github.com/marketplace/actions/setup-lazarus-environment" rel="nofollow noopener">Setup Lazarus environment</a>” action in marketplace.  Thank you for your contribution to GitHub Actions. </p>
<p><a href="https://github.com/actions/toolkit/tree/master/packages/core" rel="nofollow noopener"><strong>@actions/core</strong></a> provides core functions for setting results, logging, registering secrets and exporting variables across actions. You could see the specific functions in the readme file. </p>
<p>And <a href="https://github.com/actions/toolkit/tree/master/packages/tool-cache" rel="nofollow noopener">actions/tool-cache</a> provide functions to download and cache tools. You could use downloadTool() and extractZip() . </p>

**Comment by gcarreno** [comment]
<p>Hi Yanjingzhu,</p>
<p>I’m going to accept your answer because it’s factually correct.</p>
<p>It doesn’t answer my question, so I’ll have to Google and investigate on my own.</p>
<p>None the less, many thanks for your help!</p>
<p>Cheers,</p>
<p>Gus</p>

**Comment by chingc** [comment]
<p>Have you looked into homebrew? <a href="https://brew.sh/" rel="nofollow noopener">https://brew.sh/</a></p>
<p>It’s an extremely popular cli package manager on macos.  It also has “homebrew cask” builtin to handle the installation of applications that normally requires a “drag icon to /Application directory” type of installation.</p>
<p>I’m not familiar with the macos environments on GitHub Actions, but if homebrew is already installed then installing the software you need is as simple as</p>
<p>free pascal compiler: brew install fpc</p>
<p>pascal ide: brew cask install lazarus</p>
